x/tbr/internal/keeper/keeper.go

```diff
@@ -13,153 +12,193 @@ import (
+	// Truncate the result so that 1.99 years = 1 year and not 2
+	blocksPerYear := DPY.Mul(HPD).Mul(MPH).Mul(BPM)
+	return sdk.NewDec(blockHeight).Quo(blocksPerYear).TruncateInt64()
+}
 
-	firstMember = firstMember.Mul(secondMember)
+// SetYearNumber set the given year to be the current year number
+func (k Keeper) SetYearNumber(ctx sdk.Context, year int64) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set([]byte(types.YearNumberStoreKey), k.cdc.MustMarshalBinaryBare(year))
+}
 
-	return firstMember
+// GetYearNumber returns the current year number
+func (k Keeper) GetYearNumber(ctx sdk.Context) (year int64) {
+	store := ctx.KVStore(k.storeKey)
+	actualBz := store.Get([]byte(types.YearNumberStoreKey))
+	if actualBz == nil {
+		return 0
+	} else {
+		k.cdc.MustUnmarshalBinaryBare(actualBz, &year)
+		return year
+	}
 }
 
-// ComputeProposerReward computes the final reward for the validator block's proposer
-func (k Keeper) ComputeProposerReward(ctx sdk.Context, validatorNumber int64, proposer exported.ValidatorI,
-	totalStakedTokens sdk.Int) sdk.DecCoins {
+// UpdateYearlyPool allows to update the current yearly pool based on whenever we
+// are now in a new year or not.
+// If we are in the same year, we do nothing. Otherwise, we set the new year's pool to be the
+// 20% of the total reward pool.
+func (k Keeper) UpdateYearlyPool(ctx sdk.Context, blockHeight int64) {
+	previousYearNumber := k.GetYearNumber(ctx)
+	currentYearNumber := computeYearFromBlockHeight(blockHeight)
+
+	// Check if the year number has changed and thus we need to update the yearly reward pool
+	if previousYearNumber != currentYearNumber {
+
+		// Get the reward pool
+		rewardPool := k.GetTotalRewardPool(ctx)
+
+		// Compute a pool in which each coin is 20% of the total pool
+		yearlyRewardPool := make(sdk.DecCoins, len(rewardPool))
+		for index, coin := range rewardPool {
+			// Each new coin amount must be 20% (= 1/5) of the previous
+			yearlyRewardPool[index] = sdk.NewDecCoinFromDec(coin.Denom, coin.Amount.QuoInt64(5))
+		}
 
-	// Get the raw reward for proposer
-	rawReward := computeRawReward(validatorNumber)
+		// Set the new yearly reward pool and year number
+		k.SetYearlyRewardPool(ctx, yearlyRewardPool)
+		k.SetYearNumber(ctx, currentYearNumber)
+	}
+}
 
-	// Retrieve staked tokens by proposer
-	validatorStakedTokens := proposer.GetBondedTokens().ToDec()
+// ---------------------------
+// --- Reward distribution
+// ---------------------------
 
-	// Compute his validation power
-	validatorPower := validatorStakedTokens.Quo(totalStakedTokens.ToDec())
+// ComputeProposerReward computes the final reward for the validator block's proposer
+func (k Keeper) ComputeProposerReward(ctx sdk.Context, validatorsCount int64,
+	proposer exported.ValidatorI, totalStakedTokens sdk.Int) sdk.DecCoins {
+
+	// Get the maximum year reward by multiplying the yearly pool by V/100
+	Ry := k.GetYearlyRewardPool(ctx).MulDec(sdk.NewDec(validatorsCount)).QuoDec(sdk.NewDec(100))
```

Comment:
I think that If you use `GetYearlyRewardPool` the formula doesn't work properly: every `ComputeProposerReward` iteration gets amount under the yearly reward pool target.  In discussed formula the value for `Ry` is `Py x V/100` and it is only function of year changes and number of validators.

Reply:
You're right. I've revisited the code with the following changes: 

- Added the `YearlyPoolRemains` which tells how many coins are left for the current year
- Now the yearly pool is no more decremented when distributing the rewards. Only the `TotalRewardPool` and the `YearlyPoolRemains` are